src/main/scala/is/hail/expr/ir/Interpret.scala

```diff
@@ -408,6 +408,21 @@ object Interpret {
           null
         else
           oValue.asInstanceOf[Row].get(idx)
+      case StringSlice(s, start, end) =>
+        val Array(maybeString, vstart: Int, vend: Int) =
+          Array(s, start, end).map(interpret(_, env, args, agg))
+        if (maybeString == null)
+          null
+        else {
+          val vs = maybeString.asInstanceOf[String]
+          if (vstart < 0 || vstart == vs.length || vstart > vend || vend > vs.length)
+            fatal(s"""string slice out of bounds or invalid: "$vs"[$vstart:$vend]""")
+          else
+            vs.substring(vstart, vend)
+        }
+      case StringLength(s) =>
+        val vs = interpret(s).asInstanceOf[String]
+        if (vs == null) null else vs.getBytes().length
```

Comment:
Shouldn't this just be `vs.length()`?

Reply:
I don't know what to do exactly. In the region value we represent strings as arrays of bytes, so, for example, the poop emoji is length 4 because it needs two UTF-16 code points (each 16 bits). `String.length` is a very smart function that actually knows how unicode works.